server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -363,7 +418,8 @@ private ImmutableList<Block> getBlocks(
               blockDefinition,
               applicantData,
               maybeRepeatedEntity);
-      if (includeBlockIfTrue.test(block)) {
+      if ((includeBlockIfTrue.test(block) && !includeHidden)
+          || (!includeBlockIfTrue.test(block) && includeHidden)) {
```

Comment:
(blocking) This is a getting a little hard to follow. Can you move the check into its own method and include comments on what it's doing.

Something like

```Java
/**
 *
 */
private Boolean includeBlockInList(Predicate<Block> includeBlockIfTrue, Boolean includeHidden) {
   return (includeBlockIfTrue.test(block) && !includeHidden) || (!includeBlockIfTrue.test(block) && includeHidden);
}
```

or maybe more verbose for future readers

```Java
/**
 *
 */
private Boolean includeBlockInList(Predicate<Block> includeBlockIfTrue, Boolean includeHidden) {
  // What case does this cover?
  if (includeBlockIfTrue.test(block) && !includeHidden) {
    return true;
  }

  // What case does this cover?
  if (!includeBlockIfTrue.test(block) && includeHidden) {
    return true;
  }

  // What case does this cover?
   return false;
}
```

Reply:
+1.

Another way to simplify this might be to remove the check for `includeHidden` and flip the logic in the includeBlockIfTrue predicate. so then line 168 might look like
`allActiveBlockList = getBlocks((block) -> showBlock(block));`

but line 178 might look like
`allHiddenBlockList = getBlocks(block -> !showBlock(block));`

Reply:
Changes made. Thank you both for the suggestions!